test/central-config-enabled.js

```diff
@@ -36,15 +44,28 @@ test('remote config enabled', function (t) {
       centralConfig: true
     })
 
-    Object.defineProperty(agent._conf, 'transactionSampleRate', {
-      set (value) {
-        t.equal(value, 0.42)
+    for (const key in expect) {
+      if (!Object.prototype.hasOwnProperty.call(agent._conf, key)) {
+        t.fail('unknown config key: ' + key)
         t.end()
-      },
-      get () {},
-      enumerable: true,
-      configurable: true
-    })
+      } else {
+        Object.defineProperty(agent._conf, key, {
+          set (value) {
+            const expectValue = expect[key]
+            if (expectValue !== undefined) {
+              t.equal(value, expectValue)
+              delete expect[key]
+              if (expect === {}) {
```

Comment:
This if-statement will never evaluate true as `expect` is one object and `{}` is another - and hence never the same object. The reason why this test works anyway is that the `t.plan()` statement above ends the test automatically after 4 assertions.

You can choose to either leave it at that and remove this if-block, or you can update the if-statement to check th amount of keys left in `expect` instead, e.g:

```suggestion
              if (Object.keys(expect).length === 0) {
```

Reply:
oops, and thank you :man_facepalming: